### ISSUE TITLE
problems with bin<->det_pos

### ISSUE THREAD

**Issue by KrisThielemans** [NikEfth/STIR#26]
Like @ashgillman I have some trouble understanding `get_det_pos_pair_for_bin` and the use of `abs(timing_pos)` here
https://github.com/NikEfth/STIR/blob/7437adfbdb6578506eba59797425489a87202f60/src/include/stir/ProjDataInfoCylindricalNoArcCorr.inl#L192-L197

Note that `DetectionPositionPair` uses `unsigned int` for timing_pos`, so the `abs` is necessary, but I'd expect a swap in the detectors if it's negative.

`test_proj_data_info` checks consistency between `get_det_pos_pair_for_bin` and `get_bin_for_det_pos_pair`. However, it doesn't check if it's actually correct. I thought I'd check if the swap occurs and it doesn't seem to.

I checked this in Python, but I first had to modify `stir.i` (not committed yet)
```swig
%attributeref(stir::DetectionPosition<unsigned int>, unsigned int, tangential_coord);
%attributeref(stir::DetectionPosition<unsigned int>, unsigned int, axial_coord);
%attributeref(stir::DetectionPosition<unsigned int>, unsigned int, radial_coord);
%include "stir/DetectionPosition.h"
// Instantiate the template, but have to use a different name than the class currently I believe
%template(DetectionPositionXX) stir::DetectionPosition<unsigned int>;
#if SWIG_VERSION >= 0x040100
  %attribute2ref(stir::DetectionPositionPair<unsigned int>, DetectionPosition<unsigned int>, pos1);
   %attribute2ref(stir::DetectionPositionPair<unsigned int>, DetectionPosition<unsigned int>, pos2);
#else
// can't use the above for older swig due to typos, fixed in https://github.com/swig/swig/commit/bcfa9272980cdb99b04125f0077d9f44565bca79
  %attribute_custom(stir::DetectionPositionPair<unsigned int>, DetectionPosition<unsigned int>, pos1, pos1, pos1, &self_->pos1(), self_->pos1() = *val_);
  %attribute_custom(stir::DetectionPositionPair<unsigned int>, DetectionPosition<unsigned int>, pos2, pos2, pos2, &self_->pos2(), self_->pos2() = *val_);
#endif
%attributeref(stir::DetectionPositionPair<unsigned int>, unsigned int, timing_pos);
%include "stir/DetectionPositionPair.h"
%template(DetectionPositionPairXX) stir::DetectionPositionPair<unsigned int>;
```

```python
import stir
scanner=stir.Scanner_get_scanner_from_name("Signa PET/MR")
p=stir.ProjDataInfo.construct_proj_data_info(scanner, 1,9, 224,357,False,1)
b=stir.Bin(1,1,1,1,1)
dp=stir.DetectionPositionPairXX()
p.get_det_pos_pair_for_bin(dp,b)
print((dp.pos1.axial_coord,dp.pos2.axial_coord,dp.timing_pos))
b2=stir.Bin(1,1,1,1,-1)
dp2=stir.DetectionPositionPairXX()
p.get_det_pos_pair_for_bin(dp2,b2)
print((dp2.pos1.axial_coord,dp2.pos2.axial_coord,dp2.timing_pos))
```
Output is `(1,2,1)` in both cases, which cannot be correct.

It seems we need a C++ test for this.

However, I cannot understand why things would work reasonably currently.

@robbietuk, this might affect you when testing symmetries. No idea....

**Comment by robbietuk** [comment]
>@robbietuk, this might affect you when testing symmetries. No idea.... 

Not that I have noticed yet in https://github.com/NikEfth/STIR/pull/25

**Comment by KrisThielemans** [comment]
I've implemented the test in C++ in 8a34b4a7812ba5aff08d89c7ed77d93c42184332, and it fails. Could anyone check the logic of the test? @NikEfth @ashgillman @robbietuk

**Comment by robbietuk** [comment]
Logic looks good and test is correct in failing. I also dont quite understand `dp.timing_pos() = std::abs(bin.timing_pos_num())*this->get_tof_mash_factor();` A standard scanner ranges from `-x to x`, with 0 being center? At least thats what I find in #25.

**Comment by ashgillman** [comment]
I'd have to dig a bit further to refresh my mind. But from memory quickly now, that test is implemented how I would have expected. But, I think instead the code was implemented such that TOF Positions were always swapped such that they were always positive.

When I attempted first to merge, I found that this clashed with some of Elise's work, and attempted to change that.
See here: https://github.com/ashgillman/STIR/commit/dff2c9fc1923f1839ad879478eb7fcca5283af07
